src/Authentication/Actions/CreateIdentityTrait.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace CodeIgniter\Shield\Authentication\Actions;
+
+use CodeIgniter\Shield\Entities\User;
+use CodeIgniter\Shield\Models\DatabaseException;
+use CodeIgniter\Shield\Models\UserIdentityModel;
+
+/**
+ * Create an identity for auth action
+ */
+trait CreateIdentityTrait
+{
+    /**
+     * Create an identity for auth action
+     */
+    private function createIdentity(User $user, string $name, string $extra): string
```

Comment:
It seems this only works with actions that require a code and nothing else, as there's no way to pass either `secret` or `secret2`. I would suggest that either: 

a) the name and docblock are changed to something more specific
or
b) It accepts a $data array where the calling function can specify the bits, and then a flag that specifies whether or not to create a random string. 

My personal preference would be option b.

Reply:
This method is only for the two action classes. I've changed the method name.

If we need another method to pass different arguments, we can consider adding more methods or commonizing them at that time.